src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -0,0 +1,1042 @@
+            AssertEx.NotNull(fieldToken.Parent);
+            var info = model.GetSymbolInfo(fieldToken.Parent);
+            Assert.Empty(info.CandidateSymbols);
+            Assert.False(info.IsEmpty);
+            Assert.Equal("<P1>k__BackingField", info.Symbol!.Name);
+        }
+
+        [Fact]
+        public void TestGetFieldsToEmit()
+        {
+            var comp = CreateCompilation(@"
+public class C
+{
+    public string P { set => field = value; }
+}
+");
+            for (int i = 0; i < 3; i++)
+            {
+                var fields = ((SourceMemberContainerTypeSymbol)comp.GetTypeByMetadataName("C")!).GetFieldsToEmit().ToArray();
+                Assert.Equal(1, fields.Length);
+                Assert.Equal("<P>k__BackingField", fields[0].Name);
+            }
+        }
+
+        [Fact]
+        public void TestGetFieldsToEmit_ExpressionBodied()
+        {
+            var comp = CreateCompilation(@"
+public class C
+{
+    public string P => field;
+}
+");
+            for (int i = 0; i < 3; i++)
+            {
+                var fields = ((SourceMemberContainerTypeSymbol)comp.GetTypeByMetadataName("C")!).GetFieldsToEmit().ToArray();
+                Assert.Equal(1, fields.Length);
+                Assert.Equal("<P>k__BackingField", fields[0].Name);
+            }
+        }
+
+        [Fact]
+        public void TestGetFieldsToEmit_Initializer()
+        {
+            var comp = CreateCompilation(@"
+public class C
+{
+    public string P { get; } = string.Empty;
+}
+");
+            for (int i = 0; i < 3; i++)
+            {
+                var fields = ((SourceMemberContainerTypeSymbol)comp.GetTypeByMetadataName("C")!).GetFieldsToEmit().ToArray();
+                Assert.Equal(1, fields.Length);
+                Assert.Equal("<P>k__BackingField", fields[0].Name);
+            }
+        }
+
+        [Fact]
+        public void AssignReadOnlyOnlyPropertyOutsideConstructor()
+        {
+            var comp = CreateCompilation(@"
+class Test
+{
+    int X
+    {
+        get
+        {
+            field = 3;
+            X = 3;
+            return 0;
+        }
+    }
+}
+");
+            comp.VerifyDiagnostics(
+                // (8,13): error CS0191: A readonly field cannot be assigned to (except in a constructor or init-only setter of the type in which the field is defined or a variable initializer)
+                //             field = 3;
+                Diagnostic(ErrorCode.ERR_AssgReadonly, "field").WithLocation(8, 13),
+                // (9,13): error CS0200: Property or indexer 'Test.X' cannot be assigned to -- it is read only
+                //             X = 3;
+                Diagnostic(ErrorCode.ERR_AssgReadonlyProp, "X").WithArguments("Test.X").WithLocation(9, 13)
+            );
+        }
+
+        [Fact]
+        public void AssignReadOnlyOnlyPropertyInConstructor()
```

Comment:
>AssignReadOnlyOnlyPropertyInConstructor [](http://example.com/codeflow?start=20&length=39)

I expect this scenario to not trigger extra binding. Please add verification accordingly.